File: Kth smallest Element Leetcode 230/Solution.py
class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None
        
class Solution:
    def kthSmallest(self, root: Node, k: int) -> int:
        count = [k]
        ans = [0]
        
        def dfs(root):
            if not root:
                return
            
            dfs(root.left)
            if count[0] == 1:
                ans[0] = root.data
                
            count[0] -= 1
            
            if count[0] > 0:
                dfs(root.right)
                
        dfs(root)
        return ans[0]

File: Kth smallest Element Leetcode 230/Solution.java
class Node{
    int data;
    Node left;
    Node right;

    public Node(int data, Node left, Node right) {
        this.data = data;
        this.left = left;
        this.right = right;
    }
}
public class Solution{
    private int count;
    private int ans;

    public void dfs(Node root){
        if (root == null){
            return;
        }
        dfs(root.left);
        if (count == 1){
            ans = root.data;
        }
        count -= 1;
        if (count > 0){
            dfs(root.right);
        }
    }
    public int kthSmallestElement(Node root, int k){
        count = k;
        ans = 0;
        dfs(root);
        return ans;
    }
}